pkg/util/metric/mometric/metric_collector_log.go

```diff
@@ -0,0 +1,57 @@
+package mometric
+
+import (
+	"context"
+	"github.com/matrixorigin/matrixone/pkg/logutil"
+	pb "github.com/matrixorigin/matrixone/pkg/pb/metric"
+	bp "github.com/matrixorigin/matrixone/pkg/util/batchpipe"
+)
+
+// TODO: Once this class is mature, it will be moved inside metric_collector.go
+type metricLogCollector struct {
```

Comment:
My suggestion like this: https://github.com/xzxiong/matrixone/pull/31/files
you can do the log in function `GetBatch`